velox/exec/benchmarks/ExchangeBenchmark.cpp

```diff
@@ -56,28 +59,35 @@ using namespace facebook::velox::test;
 
 namespace {
 
-struct Counters {
-  int64_t bytes{0};
-  int64_t rows{0};
-  int64_t usec{0};
-  int64_t repartitionNanos{0};
-  int64_t exchangeNanos{0};
-  int64_t exchangeRows{0};
-  int64_t exchangeBatches{0};
-
-  std::string toString() {
-    if (exchangeBatches == 0) {
-      return "N/A";
-    }
-    return fmt::format(
-        "{}/s repartition={} exchange={} exchange batch={}",
-        succinctBytes(bytes / (usec / 1.0e6)),
-        succinctNanos(repartitionNanos),
-        succinctNanos(exchangeNanos),
-        exchangeRows / exchangeBatches);
-  }
+struct LocalPartitionWaitStats {
+  int64_t totalProducerWaitMs = 0;
+  int64_t totalConsumerWaitMs = 0;
+  std::vector<RuntimeMetric> consumerWaitMs;
+  std::vector<RuntimeMetric> producerWaitMs;
+  std::vector<int64_t> wallMs;
 };
 
+void sortByMax(std::vector<RuntimeMetric>& metrics) {
+  std::sort(
+      metrics.begin(),
+      metrics.end(),
+      [](const RuntimeMetric& left, const RuntimeMetric& right) {
+        return left.max > right.max;
+      });
+}
+
+void printMax(
+    const char* title,
+    int64_t total,
+    std::vector<RuntimeMetric>& metrics) {
+  sortByMax(metrics);
+  assert(!metrics.empty());
+  std::cout << title << " Total " << succinctNanos(total)
+            << " Max: " << metrics.front().toString()
+            << " Median: " << metrics[metrics.size() / 2].toString()
+            << " Min: " << metrics.back().toString() << std::endl;
```

Comment:
This produces summaries like this one:
```
Producer Total 0ns Max: sum:5190498000, count:1, min:5190498000, max:5190498000 Median: sum:1852680000, count:1, min:1852680000, max:1852680000 Min: sum:1477941000, count:16, min:35098000, max:137225000
```
I believe it would be much easier to read if we add more formatting and present this information in the following way:
```
Producer:
    Total: 0ns
    Max: [sum:5190498000, count:1, min:5190498000, max:5190498000]
    Median: [sum:1852680000, count:1, min:1852680000, max:1852680000]
    Min: [sum:1477941000, count:16, min:35098000, max:137225000]
```